kroxylicious-operator/src/main/java/io/kroxylicious/kubernetes/operator/ClusterCondition.java

```diff
@@ -6,22 +6,34 @@
 
 package io.kroxylicious.kubernetes.operator;
 
+import java.util.Locale;
+import java.util.Optional;
+
+import io.kroxylicious.kubernetes.api.v1alpha1.virtualkafkaclusterspec.TargetCluster;
+
+import edu.umd.cs.findbugs.annotations.NonNull;
+import edu.umd.cs.findbugs.annotations.Nullable;
+
 import static io.kroxylicious.kubernetes.api.v1alpha1.kafkaproxystatus.clusters.Conditions.Status;
 
-public record ClusterCondition(String cluster, ConditionType type, Status status, String reason, String message) {
+public record ClusterCondition(@NonNull String cluster,
+                               @NonNull ConditionType type,
+                               @NonNull Status status,
+                               @Nullable String reason,
+                               @Nullable String message) {
 
     public static final String INVALID = "Invalid";
 
     static ClusterCondition accepted(String cluster) {
         return new ClusterCondition(cluster, ConditionType.Accepted, Status.TRUE, null, null);
     }
 
-    static ClusterCondition filterInvalid(String cluster, String filterName, String reason) {
+    static ClusterCondition filterInvalid(String cluster, String filterName, String detail) {
         return new ClusterCondition(cluster, ConditionType.Accepted, Status.FALSE, INVALID,
-                String.format("Filter \"%s\" is invalid: %s", filterName, reason));
+                String.format("Filter \"%s\" is invalid: %s", filterName, detail));
     }
 
-    static ClusterCondition filterNotExists(String cluster, String filterName) {
+    static ClusterCondition filterNotFound(String cluster, String filterName) {
```

Comment:
Since we're going to adopt ResolvedRefs, would `*RefUnresolved` or `*RefNotResolved` be more in line with that terminology?